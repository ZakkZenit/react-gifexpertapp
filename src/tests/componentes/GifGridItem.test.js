import React from 'react';
import { shallow } from 'enzyme'
import { GifGridItem } from "../../componentes/GifGridItem";


describe('Pruebas en el <GifGridItem />', () => {

    const title = 'Titulo';
    const url = 'https://localhost/algo.jpg'

    let wrapper = shallow( <GifGridItem title={title} url={url} /> );
    
    beforeEach( () =>  {

        wrapper = shallow( <GifGridItem title={title} url={url}/> );

    });

    test('debe mostrar el componenete correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe tener parrafo con titulo ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('debe tener animate fadeIn', () => {
        const div = wrapper.find('div');

        expect(div.hasClass('animate__fadeIn')).toBe(true);
    })
    
    
    
});