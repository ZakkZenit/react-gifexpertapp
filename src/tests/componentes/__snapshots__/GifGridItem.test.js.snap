// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGridItem /> debe mostrar el componenete correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
